### PR TITLE
Book Store - Connect to API

### PR DESCRIPTION
**Pull request by aamirkhan2478**
## This Pull request is to connect API with react redux.
- Fetch book list from Bookstore API on Books page load.
  - Add an Action Creator, that returns a function.
  - Fetch a list of books.
  - Dispatch another action (add an action type) that will pass the API response to the reducer and update the state.
- Refactor your add book and remove book features to persist your changes in the server.
  - Change your existing Action Creators, so they return functions.
  - Make changes by using API.
  - Dispatch an action (action type that existed before refactor) that will update the state accordingly.
- Add the book and remove the book should work in the same way after the refactor.
- No styling is required.

### CODE REVIEW

File: src/Redux/Books/books.js [hunk 1]
```diff
@@ -16,32 +21,72 @@ const bookReducer = (state = initialState, action) => {
...
 
 // Actions
-export const addBook = (values) => (dispatch) => {
-  dispatch({
-    type: ADD_BOOK,
-    payload: values,
-  });
+export const showBook = () => async (dispatch) => {
+  try {
+    const { data } = await axios.get(baseUrl);
```
Comment (ShoiraTa) [inline src/Redux/Books/books.js]: - [ ] Everything in your project is working as expected. However, based on Project requirements you have to use createAsyncThunk for creating thunk action creators:  
```
// This code is just an example

import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios'; 


export const getBooks = createAsyncThunk(
  'books/getBooks',
  async () => {
    const response = await axios.get(
      'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/',
    );
    return response.data;
  },
);
```

Please make sure that thunk action creator for fetching books is  created

* It should return a promise.
* It should fetch books list from API.
* It should automatically dispatch an action type that will pass books list to reducer and updates state.

**Review by ShoiraTa — CHANGES_REQUESTED** [review]
Hi @aamirkhan2478 ,

:thumbsup: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

**First let me to point out:**

- All the functionality works as expected ✔️ 
- You passed all the Linters! ✔️ 
- Professional pull request! ✔️ 

Keep on coding, good job so far! :clap:

There are some issues that you still need to work on to go to the next project but you are almost there!



 ## Required Changes  :recycle: 

_Check the comments under the review._ 

***
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear by mentioning my Github username.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

* * *
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/Redux/Books/books.js (hunk 1 above)
Comment (aamirkhan2478) [reply src/Redux/Books/books.js:47]: Hello @ShoiraTa,
I didn't use createSlice for my project without createSlice how I can use createAsyncThunk in my project?

**Review by topeogunleye — COMMENTED** [review]
### Hi @aamirkhan2478, 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
>
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Followed Github flow :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and description :heavy_check_mark:
 
## [Optional] suggestions:
* Nothing to mention.
 
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**